pkg/locking/control_plane_init_mutex.go

```diff
@@ -0,0 +1,191 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package locking implements locking functionality.
+package locking
```

Comment:
Does this come from upstream somewhere? If it does, could we reference it instead of copying it into our project?

Reply:
This comes indeed from upstream. However, in CAPI, it is an internal package, which means it cannot be imported, but can be copied. Should we change the headers?

Reply:
Add a comment in the code saying, that this code was taken over from upstream Kubeadm.